wicker/core/datasets.py

```diff
@@ -207,6 +272,104 @@ def schema(self) -> DatasetSchema:
                 )
         return self._schema
 
+
```

Comment:
As I mentioned in the PR description, now the only thing that `FileSystemDataset` has is an `__init__` function that further constrains the types of its arguments like `storage` and `pa_filesystem`. This can help the user to reduce any type-related errors. However, I would be ok to remove this class if the reviewers do not think it is very useful.

Reply:
Another way to do things would be to remove `pa_filesystem`, `path_factory` and `storage` as arguments to the `__init__` function and instead add factory methods to the class that build an appropriate instance for each of these. This enables them to be removed from the constructor and also enables subclasses to override the factory methods in case they want to pass a more specific subtype.

Reply:
From the Ioc and testable class design. I like your implementation, I would suggest a get_filesystemdataset factory to decouple the construction detail and user interface as I suggested in the previous FileSystemDataset PR by @pickles-bread-and-butter